lib/tasks/evm_settings.rake

```diff
@@ -80,8 +82,8 @@ module EvmSettings
   private_class_method :str_to_value
 
   def self.value_to_str(value)
-    return "true"  if value  || value =~ /true/i
-    return "false" if !value || value =~ /false/i
+    return "true"  if value == true || value =~ /true/i
+    return "false" if value == false || value =~ /false/i
```

Comment:
These changes were necessary because `if value` does not do what is expected if value is a string as the below code segment depicts.

```ruby
irb(main):010:0> value="none"
=> "none"
irb(main):011:0> puts "true" if value
true
=> nil
irb(main):012:0> puts "true" if value == true
=> nil
```

Reply:
This is a pretty big change in behavior.

After this change any string which doesn't match "true" will move from displaying as "true" to "false". Are you sure this won't change anything else that uses this method? What is the use case for changing this related to OIDC?

Reply:
Since this is used only by authentication settings and you tested that these all work with the new logic, I'm okay with changing this as long as you add specs.

Adding specs will really ensure that we know what the desired behavior of this module is in the future.

Reply:
@jvlcek Can you confirm that this code is only used for external auth settings? I'm trying to understand the impact of this change. 

Also, what breaks with the old code when `value="none"`? Is it just the display of the current settings?

Reply:
@gtanzillo Thank you for the feedback. @carbonin and I had done an analysis of the code and we discovered this is only used by authentication settings.

In addition I just did a search of the source code base and the gems on a live appliance and found no other consumers of this code.

The last line of the code in question was to return the value passed in if it was not
a true, false, "true" or "false". However this line was never invoked. For example if
the string "bobs your uncle" was passed in "true" was erroneously returned when
"bobs your uncle" should have been.

```ruby
 def self.value_to_str(value)
    return "true"  if value == true || value =~ /true/i
    return "false" if value == false || value =~ /false/i
    value
  end
```

This small code segment depicts the old and new behavior.

```ruby
#!/usr/bin/env ruby


def old(value)
  return "true"  if value  || value =~ /true/i
  return "false" if !value || value =~ /false/i
  value
end
puts "+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-="
puts "  old"
puts "+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-="
puts "old  true   returned ->#{old(true)}<-"
puts "old \"true\"  returned ->#{old("true")}<-"
puts "old  false  returned ->#{old(false)}<-"
puts "old \"false\" returned ->#{old("false")}<-"
puts "old \"bobs your uncle\" returned ->#{old("bobs your uncle")}<-"
puts "+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-="

def new(value)
  return "true"  if value == true || value =~ /true/i
  return "false" if value == false || value =~ /false/i
  value
end
puts "\n+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-="
puts "  new"
puts "+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-="
puts "new  true   returned ->#{new(true)}<-"
puts "new \"true\"  returned ->#{new("true")}<-"
puts "new  false  returned ->#{new(false)}<-"
puts "new \"false\" returned ->#{new("false")}<-"
puts "new \"bobs your uncle\" returned ->#{new("bobs your uncle")}<-"
puts "+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-="
```

The output from the about code example is:
```bash
mac% ./if_true_false_string_logic.rb
+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=
  old
+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=
old  true   returned ->true<-
old "true"  returned ->true<-
old  false  returned ->false<-
old "false" returned ->true<-
old "bobs your uncle" returned ->true<-
+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=

+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=
  new
+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=
new  true   returned ->true<-
new "true"  returned ->true<-
new  false  returned ->false<-
new "false" returned ->false<-
new "bobs your uncle" returned ->bobs your uncle<-
+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=+-=
```

As this example shows this segment of code had a latent bug but was not uncovered until now because it only had every processed settings with true or false values.

The OpenID-Connect support added a new setting `provider_type` which has a string value of "oidc", "saml" or "none". It is used by the UI to display the correct authentication provider settings.  Without this change the value of `provider_type` becomes "true" which causes the UI and appliance_console to incorrectly display the configuration.

Please reach out to discuss further if this `tl;dr` description falls short.

Thank you JoeV